### PR TITLE
fix build with dependent-sum-0.7

### PR DESCRIPTION
**Pull request by int-e**
dependent-sum 0.7 removed some deprecated APIs, reexporting
`Data.GADT.Compare` from the `some` package instead. In particular,
the `(:=)` type is replaced bt `(:~:)`.

This changes the API so do a major version bump.

@ryantrinkle @3noch: This is another fallout from the dependent-sum update :-(

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,97 +1,167 @@
...
-      compiler: ": #GHC 7.10.1"
-      addons: {apt: {packages: [cabal-install-1.22,ghc-7.10.1], sources: [hvr-ghc]}}
-    - env: CABALVER=head GHCVER=head
-      compiler: ": #GHC head"
-      addons: {apt: {packages: [cabal-install-head,ghc-head], sources: [hvr-ghc]}}
-
-  allow_failures:
-    - env: CABALVER=head GHCVER=head
-
+    - compiler: ghc-8.8.1
```
Comment (3noch) [inline .travis.yml]: I think we want 8.8.3 here (and in the cabal file).

**Comment by 3noch** [comment]
I think you mean version 0.6.2.2

File: .travis.yml (hunk 1 above)
Comment (int-e) [reply .travis.yml]: `haskell-ci` doesn't seem to know about that version yet. But it's an easy change to make manually, except that I missed one spot initially. Hopefully the third attempt is finally correct.

**Comment by int-e** [comment]
Relatedly, it would be helpful to give [prim-uniq-0.1.0.1](http://hackage.haskell.org/package/prim-uniq-0.1.0.1) an upper bound for dependent-sum on Hackage (through the revision mechanism).

**Comment by int-e** [comment]
...one more update, now that the relevant version of dependent-sum is 0.7.

**Comment by int-e** [comment]
Ping, whom should I poke about this one? Both @mokus0 and @ryantrinkle are maintainers on Hackage.

File: src/Unsafe/Unique/Tag.hs [hunk 2]
```diff
@@ -11,6 +11,7 @@ import Unsafe.Unique.Prim
 import Unsafe.Coerce
 import Control.Monad.Primitive
 import Control.Monad
+import Data.Typeable ((:~:)(..))
```
Comment (int-e) [inline src/Unsafe/Unique/Tag.hs]: This is slightly inconsistent: `(:~:)(..)` is imported from `Data.Type.Equality` in `Data.Unique.Tag`.

**Pull request merged**